renormalizer/mps/mps.py

```diff
@@ -1240,6 +1240,156 @@ def full_wfn(self) -> np.array:
+        for ims, ms in enumerate(self):
+            ltensor = environ_L.GetLR("L", ims-1, self, identity,
+                    itensor=None, method="Enviro")
+            ltensor = ltensor.reshape(ltensor.shape[0], ltensor.shape[-1])
+            tensor = tensordot(ltensor, ms.conj(), ([0],[0]))
+            if ms.ndim == 3:
+                tensor = tensordot(tensor, ms, ([0],[0]))
+            elif ms.ndim == 4:
+                tensor = tensordot(tensor, ms, ([0,2],[0,2]))
+            L_component.append(tensor.transpose((0,2,1,3)))
+            
+            rtensor = environ_R.GetLR("R", ims+1, self, identity,
+                    itensor=None, method="Enviro")
+            rtensor = rtensor.reshape(rtensor.shape[0], rtensor.shape[-1])
+            tensor = tensordot(ms.conj(), rtensor, ([-1],[0]))
+            if ms.ndim == 3:
+                tensor = tensordot(tensor, ms, ([-1],[-1]))
+            elif ms.ndim == 4:
+                tensor = tensordot(tensor, ms, ([2,-1],[2,-1]))
+            R_component.append(tensor.transpose((0,2,1,3)))
+        
+        # merge two 1-orbital environment together
+        for ims in range(self.site_num):
+            tensor = L_component[ims]
+            for jms in range(ims+1, self.site_num):
+                if jms != ims+1:
+                    kms = jms - 1
+                    tensor = tensordot(tensor, self[kms].conj(), ([2],[0]))
+                    if self[kms].ndim == 3:
+                        tensor = tensordot(tensor, self[kms], ([2,3],[0,1]))
+                    elif self[kms].ndim == 4:
+                        tensor = tensordot(tensor, self[kms], ([2,3,4],[0,1,2]))
+                
+                rtensor = R_component[jms]
+                res = tensordot(tensor, rtensor,
+                        ([2,3],[0,1])).transpose(0,2,1,3)
+                rdm[(ims, jms)] = asnumpy(res.reshape(res.shape[0]*res.shape[1],-1))
+        return rdm
+    
+    def calc_orbital_entropy(self, entropy_type):
+        r""" Calculate 1-orbital, 2-orbital Von Neumann entropy
+
+            :math:`\textrm{entropy} = -Tr(\rho ln \rho)`
+            
+            where ``ln`` stands for natural logarithm.
+
+        Paramters
+        ---------
+        entropy_type : str
+            "1o" for 1-orbital entropy, "2o" for 2-orbital entropy
+        
+        Returns
+        -------
+        entropy : dict
+            the key is the index or the tuple of index of mps sites.
+        
+        """
+
+        if entropy_type == "1o":
+            rdm = self.calc_1ordm()
+        elif entropy_type == "2o":
+            rdm = self.calc_2ordm()
+        else:
+            assert False
+
+        entropy = {}
+        for key, dm in rdm.items():
+            w, v = scipy.linalg.eigh(dm)
+            w[w<=0] = 1e-100
+            entro = np.sum(-w*np.log(w))
+            entropy[key] = entro
+        return entropy
+    
+    def calc_orbital_mutual_entropy(self):
```

Comment:
I think a test case is missing for the method. `Mpo.identity(self.mol_list)` appears in the code but `mol_list` is no longer an attribute for `mp` now.

Reply:
I will add one.